source/common/router/config_impl.cc

```diff
@@ -467,6 +467,22 @@ VirtualHostImpl::VirtualClusterEntry::VirtualClusterEntry(const Json::Object& vi
   priority_ = ConfigUtility::parsePriority(virtual_cluster);
 }
 
+const VirtualHostImpl* RouteMatcher::findWildcardVirtualHost(const std::string& host) const {
+  for (const auto& iter : wildcard_virtual_host_suffixes_) {
+    const uint32_t i = iter.first;
+    const auto& wildcard_map = iter.second;
+    // >= because *.foo.com shouldn't match .foo.com.
```

Comment:
Why? I guess a different question here is what are the semantics of the wildcard here? Is *.foo.com not equal to .foo.com , in other words, * is non empty string?